src/e2e-test/java/io/cdap/plugin/salesforcemultiobjectsbatchsource/stepsdesign/DesignTimeSteps.java

```diff
@@ -0,0 +1,70 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.salesforcemultiobjectsbatchsource.stepsdesign;
+
+import io.cdap.plugin.salesforcemultiobjectsbatchsource.actions.SalesforceMultiObjectsPropertiesPageActions;
+import io.cdap.plugin.utils.enums.SObjects;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+/**
+ * Design-time steps of Salesforce MultiObjects Batch Source plugin.
+ */
+public class DesignTimeSteps {
+  String[] whiteListWithInvalidSObjectNames = {"Accounts", "blahblah"};
+  String[] blackListWithInvalidSObjectNames = {"Leadz", "blahblahblah"};
+
+  @When("select data pipeline source as SalesforceMultiObjects")
+  public void selectSalesforceMultiObjectsAsSource() {
+    SalesforceMultiObjectsPropertiesPageActions.selectSalesforceMultiObjectsSourceFromList();
+  }
+
+  @When("navigate to the SalesforceMultiObjects properties page")
+  public void navigateToSalesforceMultiObjectsPropertiesPage() {
+    SalesforceMultiObjectsPropertiesPageActions.navigateToSalesforceMultiObjectsPropertiesPage();
+  }
+
+  @When("configure SalesforceMultiObjects source with White List containing Account and Contact SObjects")
+  public void configureSalesforceMultiObjectsWithAccountAndContactInWhiteList() {
+    String[] whiteList = {SObjects.ACCOUNT.value, SObjects.CONTACT.value};
+    String[] blackList = {};
+    SalesforceMultiObjectsPropertiesPageActions.configureSalesforceMultiObjectsPlugin(
+      false, whiteList, blackList);
+  }
+
+  @When("configure SalesforceMultiObjects source with Black List containing " +
+    "Account, Contact and Opportunity SObjects")
+  public void configureSalesforceMultiObjectsWithAccountContactAndOpportunityInBlackList() {
+    String[] whiteList = {};
+    String[] blackList = {SObjects.ACCOUNT.value, SObjects.CONTACT.value, SObjects.OPPORTUNITY.value};
+    SalesforceMultiObjectsPropertiesPageActions.configureSalesforceMultiObjectsPlugin(
+      false, whiteList, blackList);
+  }
+
+  @When("configure SalesforceMultiObjects source with White List and Black List containing " +
+    "Invalid SObject names")
+  public void configureSalesforceMultiObjectsWithInvalidSObjectNames() {
+    SalesforceMultiObjectsPropertiesPageActions.configureSalesforceMultiObjectsPlugin(
+      false, whiteListWithInvalidSObjectNames, blackListWithInvalidSObjectNames);
+  }
+
+  @Then("verify invalid SObject name validation message")
+  public void verifyInvliadSObjectNameValidationMessage() {
```

Comment:
nit: Invliad -> Invalid
Also, can you separately test invalid whitelist and invalid blacklist in separate test cases?